core/subspan_view.cpp

```diff
@@ -46,10 +46,17 @@ template <class T, class Var> Variable subspan_view(Var &var, const Dim dim) {
   std::conditional_t<std::is_const_v<T>, const Var, Var> &var_ref = var;
   using MaybeConstT =
       std::conditional_t<std::is_const_v<Var>, std::add_const_t<T>, T>;
-  return makeVariable<span<MaybeConstT>>(
-      dims, var.unit(), values_view(var_ref),
-      var.hasVariances() ? variances_view(var_ref)
-                         : std::vector<span<MaybeConstT>>{}) /*LABEL_1*/;
+  auto valuesView = values_view(var_ref);
+  if (var.hasVariances()) {
+    auto variancesView = variances_view(var_ref);
+    return createVariable<span<MaybeConstT>>(
+        Dimensions{dims}, var.unit(),
+        Values(valuesView.begin(), valuesView.end()),
+        Variances(variancesView.begin(), variancesView.end()));
+  } else
+    return createVariable<span<MaybeConstT>>(
+        Dimensions{dims}, var.unit(),
+        Values(valuesView.begin(), valuesView.end()));
```

Comment:
Why use `begin` and `end` here?

Reply:
Firstly, to find out what is the return type of `values_view` I need to find the definitions of the functions 4 times - ok, it is `std::vector`, the second: it is not working if I do
```cpp
Variances(variancesView);
```
due to the absence of some constructors:
```
/home/igudich/work/scipp/scipp/core/include/scipp/core/variable_keyword_arg_constructor.h:92:3: error: constructor for 'scipp::core::Variances<std::vector<scipp::span<double, -1>, std::allocator<scipp::span<double, -1> > > &>' must explicitly initialize the base class 'std::tuple<vector<span<double, -1>, allocator<span<double, -1> > > &>' which does not have a default constructor
  Variances(Args &&... args)
```
With the `begin` and `end` everything just works smoothly.